vars/patchDockerImage.groovy

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+    String docker_image = "opensearchproject/${args.product}:${args.tag}"
+
+    sh"""
+    #!/bin/bash
+    set -e
+    set +x
+
+    docker pull ${docker_image}
+    """
+    sh """docker inspect --format '{{ index .Config.Labels "org.label-schema.version"}}' ${docker_image} > versionnumber"""
+
+    sh """docker inspect --format '{{ index .Config.Labels "org.label-schema.build-date"}}' ${docker_image} > time"""
+
+    sh """docker inspect --format '{{ index .Config.Labels "org.label-schema.description"}}' ${docker_image} > number"""
+
+    version = readFile('versionnumber').trim()
+    build_time = readFile('time').trim()
+    build_number = readFile('number').trim()
+
+    def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: "manifests/${version}/${args.product}-${version}.yml"))
+
+    artifactUrlX64 = "https://ci.opensearch.org/ci/dbc/distribution-build-${args.product}/${version}/${build_number}/linux/x64/tar/dist/${args.product}/${args.product}-${version}-linux-x64.tar.gz"
+
+    artifactUrlARM64 = "https://ci.opensearch.org/ci/dbc/distribution-build-${args.product}/${version}/${build_number}/linux/arm64/tar/dist/${args.product}/${args.product}-${version}-linux-arm64.tar.gz"
+
+    /*slice the time to get date value*/
+    build_date = build_time[0..3] + build_time[5..6] + build_time[8..9]
+
+    def build_qualifier = inputManifest.build.qualifier
```

Comment:
We have discussed to re-use the build lib that runs the exact / similar code.
Are you just copy pasting again them here?
https://github.com/opensearch-project/opensearch-build-libraries/blob/main/vars/buildDockerImage.groovy

Reply:
In our offline discussion the decision was to use this lib while adding new branches of actions if needed.